packages/google_sign_in/google_sign_in/example/ios/RunnerTests/GoogleSignInTests.m

```diff
@@ -26,62 +26,406 @@ - (void)setUp {
+                           [expectation fulfill];
+                         }];
+  [self waitForExpectationsWithTimeout:5.0 handler:nil];
+}
+
+- (void)testGetTokensNoAuthKeychainError {
+  id mockUser = OCMClassMock([GIDGoogleUser class]);
+  OCMStub([self.mockSharedInstance currentUser]).andReturn(mockUser);
+
+  id mockAuthentication = OCMClassMock([GIDAuthentication class]);
+  NSError *error = [NSError errorWithDomain:kGIDSignInErrorDomain
+                                       code:kGIDSignInErrorCodeHasNoAuthInKeychain
+                                   userInfo:nil];
+  [[mockAuthentication stub]
+      getTokensWithHandler:[OCMArg invokeBlockWithArgs:[NSNull null], error, nil]];
+  OCMStub([mockUser authentication]).andReturn(mockAuthentication);
+
+  FlutterMethodCall *methodCall = [FlutterMethodCall methodCallWithMethodName:@"getTokens"
+                                                                    arguments:nil];
+
+  XCTestExpectation *expectation = [self expectationWithDescription:@"expect result returns true"];
+  [self.plugin handleMethodCall:methodCall
+                         result:^(FlutterError *result) {
+                           XCTAssertEqualObjects(result.code, @"sign_in_required");
+                           XCTAssertEqualObjects(result.message, kGIDSignInErrorDomain);
+                           [expectation fulfill];
+                         }];
+  [self waitForExpectationsWithTimeout:5.0 handler:nil];
+}
+
+- (void)testGetTokensCancelledError {
+  id mockUser = OCMClassMock([GIDGoogleUser class]);
+  OCMStub([self.mockSharedInstance currentUser]).andReturn(mockUser);
+
+  id mockAuthentication = OCMClassMock([GIDAuthentication class]);
+  NSError *error = [NSError errorWithDomain:kGIDSignInErrorDomain
+                                       code:kGIDSignInErrorCodeCanceled
+                                   userInfo:nil];
+  [[mockAuthentication stub]
+      getTokensWithHandler:[OCMArg invokeBlockWithArgs:[NSNull null], error, nil]];
+  OCMStub([mockUser authentication]).andReturn(mockAuthentication);
+
+  FlutterMethodCall *methodCall = [FlutterMethodCall methodCallWithMethodName:@"getTokens"
+                                                                    arguments:nil];
+
+  XCTestExpectation *expectation = [self expectationWithDescription:@"expect result returns true"];
+  [self.plugin handleMethodCall:methodCall
+                         result:^(FlutterError *result) {
+                           XCTAssertEqualObjects(result.code, @"sign_in_canceled");
+                           XCTAssertEqualObjects(result.message, kGIDSignInErrorDomain);
+                           [expectation fulfill];
+                         }];
+  [self waitForExpectationsWithTimeout:5.0 handler:nil];
+}
+
+- (void)testGetTokensURLError {
+  id mockUser = OCMClassMock([GIDGoogleUser class]);
+  OCMStub([self.mockSharedInstance currentUser]).andReturn(mockUser);
```

Comment:
Don't these need stopMocking calls?

Reply:
The teardown calls `[self.mockSharedInstance stopMocking]`.
Only `OCMClassMock` objects with stubbed _class_ methods need `-stopMocking` called on them.  Stubbing on instance method calls `-stopMocking` when the instance is deallocated.  `mockSharedInstance` currently needs it because `[mockSharedInstance sharedInstance]` is mocked (though it won't be needed once I finish your injection suggestion).

Reply:
Yep, I just can't read; I saw `OCMStub` and `sharedInstance` in the same line and thought this was stubbing another class method.